internal/api/server.go

```diff
@@ -5,23 +5,33 @@ import (
 
 	"github.com/labstack/echo/v4"
 	"github.com/labstack/echo/v4/middleware"
+	"github.com/sirupsen/logrus"
 
+	ros_middleware "github.com/redhatinsights/ros-ocp-backend/internal/api/middleware"
 	"github.com/redhatinsights/ros-ocp-backend/internal/config"
 	"github.com/redhatinsights/ros-ocp-backend/internal/logging"
 )
 
+var log *logrus.Logger = logging.GetLogger()
+var cfg *config.Config = config.GetConfig()
+
 func StartAPIServer() {
 	app := echo.New()
-	log := logging.GetLogger()
-	cfg := config.GetConfig()
 	app.Use(middleware.Logger())
 	app.Use(middleware.CORSWithConfig(middleware.CORSConfig{
 		AllowMethods: []string{http.MethodGet},
 	}))
 
-	app.GET("/api/cost-management/v1/recommendations/openshift", GetRecommendationSetList)
-	app.GET("/api/cost-management/v1/recommendations/openshift/:recommendation-id", GetRecommendationSet)
-	app.File("/api/cost-management/v1/recommendations/openshift/openapi.json", "openapi.json")
+	app.GET("/status", GetAppStatus)
+
+	v1 := app.Group("/api/cost-management/v1")
+	v1.Use(ros_middleware.Identity)
+	if cfg.RBACEnabled {
+		v1.Use(ros_middleware.Rbac)
+	}
+	v1.GET("/recommendations/openshift", GetRecommendationSetList)
+	v1.GET("/recommendations/openshift/:recommendation-id", GetRecommendationSet)
+	v1.File("/recommendations/openshift/openapi.json", "openapi.json")
```

Comment:
```suggestion
	v1.File("/recommendations/openshift/openapi.json", "internal/api/docs/v1/openapi.json")
```

Getting following error above change, I think we should exclude this endpoint from rbac

`{
    "message": "Unable to marshal X-Rh-Identity into struct"
}`

Reply:
@saltgen  +1 to excluding this endpoint.  URL `{path} /openapi.json` should be externally available.

Reply:
`/openapi.json` does not require any RBAC permission but atleast user should be logged into consoledot.

Reply:
IIUC, console login should not be a necessity as well. I get valid response from below request.
`curl "https://console.redhat.com/api/policies/v1/openapi.json" | python -m json.tool `

Reply:
Ok I will exclude /openapi.json route from rbac.